servers/src/main/java/tachyon/worker/netty/protocol/EncodedMessage.java

```diff
@@ -0,0 +1,29 @@
+/*
+ * Licensed to the University of California, Berkeley under one or more contributor license
+ * agreements. See the NOTICE file distributed with this work for additional information regarding
+ * copyright ownership. The ASF licenses this file to You under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance with the License. You may obtain a
+ * copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package tachyon.worker.netty.protocol;
+
+import io.netty.buffer.ByteBuf;
+import io.netty.handler.codec.ByteToMessageDecoder;
+import io.netty.handler.codec.LengthFieldBasedFrameDecoder;
+
+public interface EncodedMessage {
+
+  // Returns the number bytes for the encoded message.
+  public int getEncodedLength();
+
+  // Writes the encoded message to the output ByteBuf.
+  public void encode(ByteBuf out);
```

Comment:
NIT: the name is confusing. When I first read this, I assume this is already an encoded message, and I was wondering what can you further encode to an encoded message?
The function of this method, however, is more like serialize something to buffer, maybe serializeToBuffer ?

It could be the case you want to keep consistency with Netty convention (it seems they use encode for this meaning). In that case, feel free to discard my comment here :)